fedimint-core/src/core.rs

```diff
@@ -240,192 +267,99 @@ macro_rules! newtype_impl_display_passthrough_with_instance_id {
     };
 }
 
-/// Module Decoder trait
-///
-/// Static-polymorphism version of [`IDecoder`]
-///
-/// All methods are static, as the decoding code is supposed to be
-/// instance-independent, at least until we start to support modules with
-/// overriden [`ModuleInstanceId`]s
-pub trait Decoder: Debug + Send + Sync + 'static {
-    type Input: Input;
-    type Output: Output;
-    type OutputOutcome: OutputOutcome;
-    type ConsensusItem: ModuleConsensusItem;
-
-    /// Decode `Input` compatible with this module, after the module key prefix
-    /// was already decoded
-    fn decode_input(&self, r: &mut dyn io::Read) -> Result<Self::Input, DecodeError>;
-
-    /// Decode `Output` compatible with this module, after the module key prefix
-    /// was already decoded
-    fn decode_output(&self, r: &mut dyn io::Read) -> Result<Self::Output, DecodeError>;
-
-    /// Decode `OutputOutcome` compatible with this module, after the module key
-    /// prefix was already decoded
-    fn decode_output_outcome(
-        &self,
-        r: &mut dyn io::Read,
-    ) -> Result<Self::OutputOutcome, DecodeError>;
-
-    /// Decode `ConsensusItem` compatible with this module, after the module key
-    /// prefix was already decoded
-    fn decode_consensus_item(
-        &self,
-        r: &mut dyn io::Read,
-    ) -> Result<Self::ConsensusItem, DecodeError>;
-}
-
-pub trait IDecoder: Debug {
-    /// Decode `Input` compatible with this module, after the module key prefix
-    /// was already decoded
-    fn decode_input(
-        &self,
-        r: &mut dyn io::Read,
-        instance_id: ModuleInstanceId,
-    ) -> Result<DynInput, DecodeError>;
-
-    /// Decode `Output` compatible with this module, after the module key prefix
-    /// was already decoded
-    fn decode_output(
-        &self,
-        r: &mut dyn io::Read,
-        instance_id: ModuleInstanceId,
-    ) -> Result<DynOutput, DecodeError>;
+/// A type that has a `Dyn*`, type erased version of itself
+pub trait IntoDynInstance {
+    /// The type erased version of this type
```

Comment:
when I read this initially, it was a bit unclear to me what `this` refers to.
nit: `/// The type erased version of the type implementing this trait`

Reply:
fixed